### ISSUE TITLE
Loss values are high

### ISSUE THREAD

**Issue by tyasird** [bowang-lab/BIONIC#57]
Hi,

I have 2 gene co-expression network (based on correlation score) and wanted to integrate them, the problem is I have tried to change couple of hyperparameter settings but still total loss values are so high. (I have uploaded my inputs)

[p.txt](https://github.com/bowang-lab/BIONIC/files/13687159/p.txt)

[e.txt](https://github.com/bowang-lab/BIONIC/files/13687156/e.txt)

my config file is:

```
{
	"net_names": [
		"input/ep/e.txt",
		"input/ep/p.txt"
	],
	"out_name": "output/first_try/",
	"save_model": true,
	"model_parallel": true,

	"epochs": 3000,
	"batch_size": 100, 
	"learning_rate": 0.000000005,
	"gat_shapes": {
		"dimension": 60,
		"n_heads": 10,
		"n_layers": 3
	},
	"embedding_size": 512,
	  "tensorboard": {
		"training": false,
		"embedding": false,
		"comment": "tensorboard_example"
	  }
}

```

`Preprocessing finished: 682 total nodes.
`

```

Epoch: 393 | Loss Total: 75780271980419072.000000 | Loss 1: 37890135990209536.000000 | Loss 2: 37890135990209536.000000 | Time (s): 0.2606
Epoch: 394 | Loss Total: 14149534704.985336 | Loss 1: 7074767352.492668 | Loss 2: 7074767352.492668 | Time (s): 0.2584
Epoch: 395 | Loss Total: 2364943089135.483887 | Loss 1: 1182471544567.741943 | Loss 2: 1182471544567.741943 | Time (s): 0.2996
Epoch: 396 | Loss Total: 8536677799.413489 | Loss 1: 4268338899.706745 | Loss 2: 4268338899.706745 | Time (s): 0.2866
Epoch: 397 | Loss Total: 19484690017.595306 | Loss 1: 9742345008.797653 | Loss 2: 9742345008.797653 | Time (s): 0.2406
Epoch: 398 | Loss Total: 76221207573.020523 | Loss 1: 38110603786.510262 | Loss 2: 38110603786.510262 | Time (s): 0.2572
Epoch: 399 | Loss Total: 7722696434937.243164 | Loss 1: 3861348217468.621582 | Loss 2: 3861348217468.621582 | Time (s): 0.2749
Epoch: 400 | Loss Total: 2096474884354.252197 | Loss 1: 1048237442177.126099 | Loss 2: 1048237442177.126099 | Time (s): 0.2755
Epoch: 401 | Loss Total: 828078994092.668579 | Loss 1: 414039497046.334290 | Loss 2: 414039497046.334290 | Time (s): 0.2976
```

**Comment by tyasird** [comment]
I found the problem, unlike SNF instead of putting whole network, I guess we need to put just significant edges.

**Comment by duncster94** [comment]
Hi, yes you should threshold your networks first. I also noticed the loss for both networks is the same - are `e.txt` and `p.txt` duplicates?

**Comment by tyasird** [comment]
they are not duplicates, same matrix shape, same edges with different correlation scores (without any thresholding). With thresholding it works good. Maybe for the future reference would be good to put some notification if both networks have the same edges.

btw, I compared with SNF and saw that clustering performance is a bit behind of SNF for human networks but still better than separated networks.

Thanks!

**Comment by duncster94** [comment]
Are your losses different after thresholding?

Yes, 682 nodes is not many for a network so in this case SNF will likely perform well.

**Comment by duncster94** [comment]
Closing due to staleness.
